providers/oidc.go

```diff
@@ -101,24 +101,33 @@ func (p *OIDCProvider) EnrichSession(ctx context.Context, s *sessions.SessionSta
 	return nil
 }
 
-// ValidateSession checks that the session's IDToken is still valid
+// ValidateSession checks that the session's id_token or access_token (when a ValidateURL is configured) is still valid
 func (p *OIDCProvider) ValidateSession(ctx context.Context, s *sessions.SessionState) bool {
-	_, err := p.Verifier.Verify(ctx, s.IDToken)
```

Comment:
So I propose that we just check both:

```golang
func (p *OIDCProvider) ValidateSession(ctx context.Context, s *sessions.SessionState) bool {
	if !validateToken(ctx, p, s.AccessToken, makeOIDCHeader(s.AccessToken)) {
		logger.Errorf("access token is invalid")
		return false
	}

	if s.IDToken == "" {
		logger.Println("no new id token has been redeemed after refresh")
		return true
	}

	if _, err := p.Verifier.Verify(ctx, s.IDToken); err != nil {
		logger.Errorf("id_token verification failed: %v", err)
		return false
	}

	if p.SkipNonce {
		return true
	}

	if err := p.checkNonce(s); err != nil {
		logger.Errorf("nonce verification failed: %v", err)
		return false
	}

	return true
}
```

Reply:
I think your snippet will break existing OICD providers:

```go
if !validateToken(ctx, p, s.AccessToken, makeOIDCHeader(s.AccessToken)) {
	logger.Errorf("access token is invalid")
	return false
}
```

`validateToken` returns `false` when `p.Data().ValidateURL == nil || p.Data().ValidateURL.String() == ""` which is the case unless you explicitly set a `ValidateURL`.

I propose we:

1. validate the ID token is it is present
2. _or_ validate the access token
3. _or_ throw an error when neither can be verified

@tuunit, what do you think?

Reply:
There seems to be one more problem, `if s.IDToken == ""` likely also does not work as the session should contain the ID token from the initial redeem. (see `redeemRefreshToken()`)

Reply:
I see the following options right now:

1. We add a configuration flag to validate the access token or even entirely disable session validation
2. We check the access token when the ID token is expired
3. We disable session validation after a refresh only (what is the added value of the validation anyways? we just successfully redeemed a refresh token - how can the session be invalid?)